### ISSUE TITLE
Cannot build or test on Fedora 36

### ISSUE THREAD

**Issue by nickbreen** [bazelbuild/rules_docker#2136]
https://github.com/bazelbuild/rules_docker/issues/1997 - there's a stale issue.

# 🐞 bug report

### Description

Cannot build on Fedora 36. Cannot build with `run-bazel-in-docker.sh` on Fedora 36.

## 🔬 Minimal Reproduction

1. Clone repo and checkout tag `v0.25.0`
3. Execute any `bazel build` or `bazel test` command.
4. Execute `./run-bazel-in-docker.sh`
5. Execute `testing/e2e.sh`

### `run-bazel-in-docker.sh`

```
$ ./run-bazel-in-docker.sh 
Unable to find image 'l.gcr.io/google/bazel:latest' locally
latest: Pulling from google/bazel
451882dd93ab: Pull complete 
208ab81fda08: Pull complete 
3c2cba919283: Pull complete 
e7fff65e05cc: Pull complete 
142dae739504: Pull complete 
Digest: sha256:ace9881e6e9c5d48b5fd637321361aeffe54000265894a65f7d818dc1065bd80
Status: Downloaded newer image for l.gcr.io/google/bazel:latest
FATAL: mkdir('/tmp/build_output'): (error: 13): Permission denied
```

```
$ ls -la /tmp/build_output/
total 0
drwxr-xr-x.  2 breenn users   40 Jul 30 10:50 .
drwxrwxrwt. 34 root   root  1280 Jul 30 11:06 ..
```

```
$ ./testing/e2e.sh
[deletions elided]
Unable to find image 'registry:2' locally
2: Pulling from library/registry
530afca65e2e: Pulling fs layer
d450d4da0343: Pulling fs layer
96277bea17b6: Pulling fs layer
470ad04e03fb: Pulling fs layer
bd3d4dc6e66f: Pulling fs layer
470ad04e03fb: Waiting
bd3d4dc6e66f: Waiting
d450d4da0343: Verifying Checksum
d450d4da0343: Download complete
530afca65e2e: Verifying Checksum
530afca65e2e: Download complete
96277bea17b6: Verifying Checksum
96277bea17b6: Download complete
470ad04e03fb: Verifying Checksum
470ad04e03fb: Download complete
bd3d4dc6e66f: Download complete
530afca65e2e: Pull complete
d450d4da0343: Pull complete
96277bea17b6: Pull complete
470ad04e03fb: Pull complete
bd3d4dc6e66f: Pull complete
Digest: sha256:c631a581c6152f5a4a141a974b74cf308ab2ee660287a3c749d88e0b536c0c20
Status: Downloaded newer image for registry:2
./testing/e2e.sh: line 110: bazel: command not found

$ PATH=$PWD:$PATH ./testing/e2e.sh
"docker rmi" requires at least 1 argument.
See 'docker rmi --help'.

Usage:  docker rmi [OPTIONS] IMAGE [IMAGE...]

Remove one or more images
DEBUG: /home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/bazel_toolchains/rules/rbe_repo/version_check.bzl:68:14: 
Current running Bazel is ahead of bazel-toolchains repo. Please update your pin to bazel-toolchains repo in your WORKSPACE file.
DEBUG: /home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/bazel_toolchains/rules/rbe_repo/checked_in.bzl:125:14: buildkite_config not using checked in configs; Bazel version 5.1.1 was picked/selected but no checked in config was found in map {"0.20.0": ["8.0.0"], "0.21.0": ["8.0.0"], "0.22.0": ["8.0.0", "9.0.0"], "0.23.0": ["8.0.0", "9.0.0"], "0.23.1": ["8.0.0", "9.0.0"], "0.23.2": ["9.0.0"], "0.24.0": ["9.0.0"], "0.24.1": ["9.0.0"], "0.25.0": ["9.0.0"], "0.25.1": ["9.0.0"], "0.25.2": ["9.0.0"], "0.26.0": ["9.0.0"], "0.26.1": ["9.0.0"], "0.27.0": ["9.0.0"], "0.27.1": ["9.0.0"], "0.28.0": ["9.0.0"], "0.28.1": ["9.0.0"], "0.29.0": ["9.0.0"], "0.29.1": ["9.0.0", "10.0.0"], "1.0.0": ["9.0.0", "10.0.0"], "1.0.1": ["10.0.0"], "1.1.0": ["10.0.0"], "1.2.0": ["10.0.0"], "1.2.1": ["10.0.0"], "2.0.0": ["10.0.0"], "2.1.0": ["10.0.0"], "2.1.1": ["10.0.0", "11.0.0"], "2.2.0": ["11.0.0"], "3.0.0": ["11.0.0"], "3.1.0": ["11.0.0"], "3.2.0": ["11.0.0"], "3.3.0": ["11.0.0"], "3.3.1": ["11.0.0"], "3.4.1": ["11.0.0"], "3.5.0": ["11.0.0"], "3.5.1": ["11.0.0"], "3.6.0": ["11.0.0"], "3.7.0": ["11.0.0"], "3.7.1": ["11.0.0"], "3.7.2": ["11.0.0"], "4.0.0": ["11.0.0"]}
ERROR: /home/breenn/src/bazelbuild/rules_docker/tests/container/BUILD:79:16: While resolving toolchains for target //tests/container:set_cmd: no matching toolchains found for types //toolchains/docker:toolchain_type
ERROR: Analysis of target '//tests/container:push_test' failed; build aborted: 
INFO: Elapsed time: 0.208s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (10 packages loaded, 511 targets configured)
    currently loading: @bazel_tools//tools/osx
```

### `bazel test`

```
$ ./bazel test //:all
DEBUG: /home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/bazel_toolchains/rules/rbe_repo/version_check.bzl:68:14: 
Current running Bazel is ahead of bazel-toolchains repo. Please update your pin to bazel-toolchains repo in your WORKSPACE file.
DEBUG: /home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/bazel_toolchains/rules/rbe_repo/checked_in.bzl:125:14: buildkite_config not using checked in configs; Bazel version 5.1.1 was picked/selected but no checked in config was found in map {"0.20.0": ["8.0.0"], "0.21.0": ["8.0.0"], "0.22.0": ["8.0.0", "9.0.0"], "0.23.0": ["8.0.0", "9.0.0"], "0.23.1": ["8.0.0", "9.0.0"], "0.23.2": ["9.0.0"], "0.24.0": ["9.0.0"], "0.24.1": ["9.0.0"], "0.25.0": ["9.0.0"], "0.25.1": ["9.0.0"], "0.25.2": ["9.0.0"], "0.26.0": ["9.0.0"], "0.26.1": ["9.0.0"], "0.27.0": ["9.0.0"], "0.27.1": ["9.0.0"], "0.28.0": ["9.0.0"], "0.28.1": ["9.0.0"], "0.29.0": ["9.0.0"], "0.29.1": ["9.0.0", "10.0.0"], "1.0.0": ["9.0.0", "10.0.0"], "1.0.1": ["10.0.0"], "1.1.0": ["10.0.0"], "1.2.0": ["10.0.0"], "1.2.1": ["10.0.0"], "2.0.0": ["10.0.0"], "2.1.0": ["10.0.0"], "2.1.1": ["10.0.0", "11.0.0"], "2.2.0": ["11.0.0"], "3.0.0": ["11.0.0"], "3.1.0": ["11.0.0"], "3.2.0": ["11.0.0"], "3.3.0": ["11.0.0"], "3.3.1": ["11.0.0"], "3.4.1": ["11.0.0"], "3.5.0": ["11.0.0"], "3.5.1": ["11.0.0"], "3.6.0": ["11.0.0"], "3.7.0": ["11.0.0"], "3.7.1": ["11.0.0"], "3.7.2": ["11.0.0"], "4.0.0": ["11.0.0"]}
INFO: Repository io_bazel_rules_docker_pip_deps instantiated at:
  /home/breenn/src/bazelbuild/rules_docker/WORKSPACE:45:8: in <toplevel>
  /home/breenn/src/bazelbuild/rules_docker/repositories/py_repositories.bzl:34:20: in py_deps
  /home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip.bzl:53:19: in pip_install
Repository rule pip_repository defined at:
  /home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip_install/pip_repository.bzl:67:33: in <toplevel>
ERROR: An error occurred during the fetch of repository 'io_bazel_rules_docker_pip_deps':
   Traceback (most recent call last):
	File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip_install/pip_repository.bzl", line 63, column 13, in _pip_repository_impl
		fail("rules_python_external failed: %s (%s)" % (result.stdout, result.stderr))
Error in fail: rules_python_external failed: Collecting six==1.11.0
  Using cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
  Saved ./six-1.11.0-py2.py3-none-any.whl
Collecting addict==2.1.2
  Using cached https://files.pythonhosted.org/packages/62/55/bbfdbdd6c44c78a0715d5569d579497cd41efa75b6ac84d6939c67e6f870/addict-2.1.2.tar.gz
Collecting PyYAML==5.4
  Using cached https://files.pythonhosted.org/packages/b5/fd/15638de2da0a5aa91c095718444624aa565f766fc178249ca6faa372f71a/PyYAML-5.4.tar.gz
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
 (ERROR: Exception:
Traceback (most recent call last):
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/cli/base_command.py", line 153, in _main
    status = self.run(options, args)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/commands/wheel.py", line 158, in run
    resolver.resolve(requirement_set)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/legacy_resolve.py", line 201, in resolve
    self._resolve_one(requirement_set, req)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/legacy_resolve.py", line 365, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/legacy_resolve.py", line 312, in _get_abstract_dist_for
    abstract_dist = self.preparer.prepare_linked_requirement(
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/operations/prepare.py", line 223, in prepare_linked_requirement
    abstract_dist = _get_prepared_distribution(
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/operations/prepare.py", line 49, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(finder, build_isolation)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/distributions/source/legacy.py", line 37, in prepare_distribution_metadata
    self._setup_isolation(finder)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/distributions/source/legacy.py", line 90, in _setup_isolation
    reqs = backend.get_requires_for_build_wheel()
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_vendor/pep517/wrappers.py", line 151, in get_requires_for_build_wheel
    return self._call_hook('get_requires_for_build_wheel', {
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_vendor/pep517/wrappers.py", line 255, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pip._vendor.pep517.wrappers.BackendUnavailable: Traceback (most recent call last):
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_vendor/pep517/_in_process.py", line 63, in _build_backend
    obj = import_module(mod_path)
  File "/usr/lib64/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'setuptools'

Traceback (most recent call last):
  File "/usr/lib64/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip_install/extract_wheels/__main__.py", line 5, in <module>
    main()
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip_install/extract_wheels/__init__.py", line 87, in main
    subprocess.run(pip_args, check=True)
  File "/usr/lib64/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'pip', 'wheel', '-r', '/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/io_bazel_rules_docker/repositories/requirements-pip.txt']' returned non-zero exit status 2.
)
ERROR: /home/breenn/src/bazelbuild/rules_docker/WORKSPACE:45:8: fetching pip_repository rule //external:io_bazel_rules_docker_pip_deps: Traceback (most recent call last):
	File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip_install/pip_repository.bzl", line 63, column 13, in _pip_repository_impl
		fail("rules_python_external failed: %s (%s)" % (result.stdout, result.stderr))
Error in fail: rules_python_external failed: Collecting six==1.11.0
  Using cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
  Saved ./six-1.11.0-py2.py3-none-any.whl
Collecting addict==2.1.2
  Using cached https://files.pythonhosted.org/packages/62/55/bbfdbdd6c44c78a0715d5569d579497cd41efa75b6ac84d6939c67e6f870/addict-2.1.2.tar.gz
Collecting PyYAML==5.4
  Using cached https://files.pythonhosted.org/packages/b5/fd/15638de2da0a5aa91c095718444624aa565f766fc178249ca6faa372f71a/PyYAML-5.4.tar.gz
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
 (ERROR: Exception:
Traceback (most recent call last):
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/cli/base_command.py", line 153, in _main
    status = self.run(options, args)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/commands/wheel.py", line 158, in run
    resolver.resolve(requirement_set)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/legacy_resolve.py", line 201, in resolve
    self._resolve_one(requirement_set, req)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/legacy_resolve.py", line 365, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/legacy_resolve.py", line 312, in _get_abstract_dist_for
    abstract_dist = self.preparer.prepare_linked_requirement(
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/operations/prepare.py", line 223, in prepare_linked_requirement
    abstract_dist = _get_prepared_distribution(
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/operations/prepare.py", line 49, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(finder, build_isolation)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/distributions/source/legacy.py", line 37, in prepare_distribution_metadata
    self._setup_isolation(finder)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/distributions/source/legacy.py", line 90, in _setup_isolation
    reqs = backend.get_requires_for_build_wheel()
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_vendor/pep517/wrappers.py", line 151, in get_requires_for_build_wheel
    return self._call_hook('get_requires_for_build_wheel', {
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_vendor/pep517/wrappers.py", line 255, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pip._vendor.pep517.wrappers.BackendUnavailable: Traceback (most recent call last):
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_vendor/pep517/_in_process.py", line 63, in _build_backend
    obj = import_module(mod_path)
  File "/usr/lib64/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'setuptools'

Traceback (most recent call last):
  File "/usr/lib64/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip_install/extract_wheels/__main__.py", line 5, in <module>
    main()
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip_install/extract_wheels/__init__.py", line 87, in main
    subprocess.run(pip_args, check=True)
  File "/usr/lib64/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'pip', 'wheel', '-r', '/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/io_bazel_rules_docker/repositories/requirements-pip.txt']' returned non-zero exit status 2.
)
ERROR: /home/breenn/src/bazelbuild/rules_docker/BUILD:51:15: //:structure_test_at_workspace_root.image depends on //testdata:link_with_files_base in repository @ which failed to fetch. no such package '@io_bazel_rules_docker_pip_deps//': rules_python_external failed: Collecting six==1.11.0
  Using cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
  Saved ./six-1.11.0-py2.py3-none-any.whl
Collecting addict==2.1.2
  Using cached https://files.pythonhosted.org/packages/62/55/bbfdbdd6c44c78a0715d5569d579497cd41efa75b6ac84d6939c67e6f870/addict-2.1.2.tar.gz
Collecting PyYAML==5.4
  Using cached https://files.pythonhosted.org/packages/b5/fd/15638de2da0a5aa91c095718444624aa565f766fc178249ca6faa372f71a/PyYAML-5.4.tar.gz
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
 (ERROR: Exception:
Traceback (most recent call last):
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/cli/base_command.py", line 153, in _main
    status = self.run(options, args)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/commands/wheel.py", line 158, in run
    resolver.resolve(requirement_set)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/legacy_resolve.py", line 201, in resolve
    self._resolve_one(requirement_set, req)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/legacy_resolve.py", line 365, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/legacy_resolve.py", line 312, in _get_abstract_dist_for
    abstract_dist = self.preparer.prepare_linked_requirement(
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/operations/prepare.py", line 223, in prepare_linked_requirement
    abstract_dist = _get_prepared_distribution(
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/operations/prepare.py", line 49, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(finder, build_isolation)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/distributions/source/legacy.py", line 37, in prepare_distribution_metadata
    self._setup_isolation(finder)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_internal/distributions/source/legacy.py", line 90, in _setup_isolation
    reqs = backend.get_requires_for_build_wheel()
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_vendor/pep517/wrappers.py", line 151, in get_requires_for_build_wheel
    return self._call_hook('get_requires_for_build_wheel', {
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_vendor/pep517/wrappers.py", line 255, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pip._vendor.pep517.wrappers.BackendUnavailable: Traceback (most recent call last):
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/pypi__pip/pip/_vendor/pep517/_in_process.py", line 63, in _build_backend
    obj = import_module(mod_path)
  File "/usr/lib64/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'setuptools'

Traceback (most recent call last):
  File "/usr/lib64/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip_install/extract_wheels/__main__.py", line 5, in <module>
    main()
  File "/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/rules_python/python/pip_install/extract_wheels/__init__.py", line 87, in main
    subprocess.run(pip_args, check=True)
  File "/usr/lib64/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'pip', 'wheel', '-r', '/home/breenn/.cache/bazel/_bazel_breenn/7c43650276da73b738d0e7f5b291bc15/external/io_bazel_rules_docker/repositories/requirements-pip.txt']' returned non-zero exit status 2.
)
ERROR: Analysis of target '//:structure_test_at_workspace_root.image' failed; build aborted: Analysis failed
INFO: Elapsed time: 4.294s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (2 packages loaded, 17 targets configured)
FAILED: Build did NOT complete successfully (2 packages loaded, 17 targets configured)
    currently loading: testdata

```

## 🌍  Your Environment

**Operating System:**

```
$ cat /etc/os-release 
NAME="Fedora Linux"
VERSION="36 (Workstation Edition)"
ID=fedora
VERSION_ID=36
VERSION_CODENAME=""
PLATFORM_ID="platform:f36"
PRETTY_NAME="Fedora Linux 36 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:36"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f36/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=36
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=36
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation

$ uname -a
Linux leaf.local 5.18.13-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Jul 22 14:03:36 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `bazel version`:**

```
$ ./bazel version
Build label: 5.1.1
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Fri Apr 8 15:49:48 2022 (1649432988)
Build timestamp: 1649432988
Build timestamp as int: 1649432988
```

**Rules_docker version: `0.25.0`**

**Anything else relevant?**

No instructions provided for build requirements.

Was unpleasantly surprised when `test/e2e.sh` _deleted_ all _my_ containers.

**Comment by nickbreen** [comment]
The same error presents with bazel versions: 4.0.0, 4.2.2, 5.1.1, 5.2.0

**Comment by nickbreen** [comment]
This appears to be resolved somehow with bazel 6.0.0.
